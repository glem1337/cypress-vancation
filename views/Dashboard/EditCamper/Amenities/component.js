import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import {
 Col, Divider, Form, Row, Skeleton,
} from 'antd';

import { CAMPER_AMENITIES_CONFIG } from 'constants/camperAmenities';

import EditCamperLayout from 'views/layouts/EditCamper';

import CustomAccommodationSection from 'views/AddNewCamper/Amenities/CustomAccommodationSection';
import AmenitySection from 'views/AddNewCamper/Amenities/AmenitySection';

import HealthAndSafetySection from 'views/Dashboard/EditCamper/Amenities/HealthAndSafetySection';

const Amenities = ({
  isValid,
  isLoading,
  handleValidateOnSubmit,
  removeCustomAccommodation,
  addCustomAccommodation,
  handleSwitchChange,
  values: { amenities, amenityHealthSafetyItems },
  onMaxAccommodationQuantityCallback,
  onMaxCustomAccommodationQuantityCallback,
  isCamperExist,
  leavePagePrepare,
}) => (
  <EditCamperLayout
    hasFooter
    canSave={isValid && !isLoading}
    onSave={handleValidateOnSubmit}
    isLoading={isLoading}
    leavePageMethod={leavePagePrepare}
  >
    <div className="container">
      <Form>
        <Row gutter={24}>
          <Col span={24}>
            <h1 className="text-headline mb-8">
              <FormattedMessage id="addNewCamper.amenities.title" />
            </h1>
            {isCamperExist && amenities ? (
              <>
                {amenities.map((amenity, index, array) => (
                  <React.Fragment key={amenity.id}>
                    <AmenitySection
                      index={index}
                      title={amenity.title}
                      icon={amenity.iconUrl}
                      handleSwitchChange={handleSwitchChange}
                      onMaxAccommodationQuantityCallback={
                        onMaxAccommodationQuantityCallback
                      }
                      items={amenity.configurationSubAmenities}
                      options={amenity.configurationAmenityOptions}
                      withDivider={index < array.length - 1}
                    />
                    {amenity.title
                      === CAMPER_AMENITIES_CONFIG.ACCOMMODATION_TITLE && (
                      <>
                        <CustomAccommodationSection
                          amenityIndex={index}
                          items={amenity.configurationCustomAmenities}
                          addCustomAccommodation={addCustomAccommodation}
                          removeCustomAccommodation={removeCustomAccommodation}
                          onMaxCustomAccommodationQuantityCallback={
                            onMaxCustomAccommodationQuantityCallback
                          }
                        />
                        <Divider className="mt-24" />
                      </>
                    )}
                  </React.Fragment>
                ))}
                <Divider />
                <HealthAndSafetySection items={amenityHealthSafetyItems} />
              </>
            ) : (
              <Skeleton active />
            )}
          </Col>
        </Row>
      </Form>
    </div>
  </EditCamperLayout>
);

Amenities.propTypes = {
  values: PropTypes.shape().isRequired,
  removeCustomAccommodation: PropTypes.func.isRequired,
  addCustomAccommodation: PropTypes.func.isRequired,
  handleValidateOnSubmit: PropTypes.func.isRequired,
  handleSwitchChange: PropTypes.func.isRequired,
  onMaxAccommodationQuantityCallback: PropTypes.func.isRequired,
  onMaxCustomAccommodationQuantityCallback: PropTypes.func.isRequired,
  isValid: PropTypes.bool.isRequired,
  isCamperExist: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  leavePagePrepare: PropTypes.func.isRequired,
};

export default Amenities;
