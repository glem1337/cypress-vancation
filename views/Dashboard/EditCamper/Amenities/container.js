import React from 'react';
import PropTypes from 'prop-types';
import { withFormik } from 'formik';
import { compose } from 'ramda';
import { v4 as uuid } from 'uuid';
import { connect } from 'react-redux';
import yup from 'lib/yupLocalised';
import { withRouter } from 'next/router';

import { handleSubmitWithProps } from 'utils/form/handleSubmit';
import isPresent from 'utils/isPresent';
import { createRouteFromPathname } from 'utils/createRouteHelper';
import { leavePagePrepare } from 'utils/camper/leavePageHelper';

import ROUTES from 'constants/routes';
import {
  AMENITIES_VALIDATION_SCHEMA,
  CAMPER_AMENITIES_CONFIG,
  CAMPER_AMENITIES_INCLUSIONS,
  FETCH_CAMPER_AMENITIES_INCLUSIONS,
} from 'constants/camperAmenities';
import { CAMPER_INCLUSION } from 'constants/camper';

import {
  fetchCamperAmenities,
  fetchCamper,
  createCamperAmenityHealthSafeties,
} from 'state/concepts/camper/actions';
import { fetchHealthAndSafetyConfig } from 'state/concepts/health-and-safety/actions';
import { loadingSelector } from 'state/data/selectors';
import {
  camperPossibleAmenitiesSelector,
  isCamperExistSelector,
} from 'state/concepts/camper/selectors';
import { amenityHealthAndSafetyItemsSelector } from 'state/concepts/health-and-safety/selectors';
import {
  createCamperAmenitiesEndpoint,
  createCamperAmenityHealthSafetiesEndpoint,
} from 'state/concepts/camper/endpoints';

import {
  hideModal as hideModalAction,
  showModal as showModalAction,
} from 'state/modal/actions';

import { dataDeleteEntity } from 'state/data/actions';

import { MESSAGE_TYPE } from 'state/flash-messages/messagesTypes';
import { showMessage } from 'state/flash-messages/actions';

import AmenitiesComponent from './component';

class Amenities extends React.Component {
  static defaultProps = {
    isLoadingAmenities: false,
    isLoadingHealthSafeties: false,
  };

  static propTypes = {
    router: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }).isRequired,
    camperId: PropTypes.string.isRequired,
    setFieldValue: PropTypes.func.isRequired,
    setFormikState: PropTypes.func.isRequired,
    showMessage: PropTypes.func.isRequired,
    showModal: PropTypes.func.isRequired,
    hideModal: PropTypes.func.isRequired,
    isValid: PropTypes.bool.isRequired,
    values: PropTypes.shape().isRequired,
    errors: PropTypes.shape().isRequired,
    touched: PropTypes.shape().isRequired,
    handleSubmit: PropTypes.func.isRequired,
    isLoadingAmenities: PropTypes.bool,
    isLoadingHealthSafeties: PropTypes.bool,
  };

  leavePage = (withSaving, redirectRoute) => async () => {
    const { hideModal, setFieldValue, router } = this.props;

    if (withSaving) {
      await setFieldValue('redirectRoute', redirectRoute);

      await this.handleValidateOnSubmit();
    } else {
      router.push(redirectRoute);
    }

    hideModal();
  };

  leavePagePrepare = leavePagePrepare.bind(this);

  static getInitialProps = async (ctx) => {
    const camperId = ctx.query.camper;

    ctx.store.dispatch(
      dataDeleteEntity({
        kind: 'amenity',
      }),
    );

    ctx.store.dispatch(
      dataDeleteEntity({
        kind: 'amenityHealthSafetyItem',
      }),
    );

    ctx.store.dispatch(
      fetchCamperAmenities(FETCH_CAMPER_AMENITIES_INCLUSIONS.join(',')),
    );

    const camperInclusions = [
      ...CAMPER_AMENITIES_INCLUSIONS,
      CAMPER_INCLUSION.CAMPER_ADDITION.AMENITY_HEALTH_SAFETY_ITEMS,
    ];

    ctx.store.dispatch(fetchCamper(camperId, camperInclusions.join(',')));
    ctx.store.dispatch(fetchHealthAndSafetyConfig());

    return { camperId };
  };

  addCustomAccommodation = (amenityIndex) => () => {
    const { setFieldValue, values } = this.props;

    const id = uuid();

    const amenities = [...values.amenities];

    amenities[amenityIndex].configurationCustomAmenities = amenities[amenityIndex]
      .configurationCustomAmenities || [];

    if (
      amenities[amenityIndex].configurationCustomAmenities.length
      === CAMPER_AMENITIES_CONFIG.MAX_CUSTOM_ACCOMMODATIONS
    ) {
      this.props.showMessage({
        messageSubTitle: {
          id: 'validations.maxAllowedNumber',
        },
        messageType: MESSAGE_TYPE.WARN,
      });

      return;
    }

    amenities[amenityIndex].configurationCustomAmenities.push({
      id,
      name: '',
      quantity: 1,
    });

    setFieldValue('amenities', amenities);
  };

  removeCustomAccommodation =
    ({ amenityIndex, index }) => () => {
      const { setFieldValue, values } = this.props;

      const amenities = [...values.amenities];
      amenities[amenityIndex].configurationCustomAmenities.splice(index, 1);

      setFieldValue('amenities', amenities);
    };

  handleSwitchChange =
    ({ amenityIndex, optionIndex }) => (isChecked) => {
      if (isChecked) return;

      const { values, errors, touched, setFormikState } = this.props;

      const { configurationSubAmenities } = values.amenities[amenityIndex]
        .configurationAmenityOptions[optionIndex];

      configurationSubAmenities.forEach((_, index) => {
        configurationSubAmenities[index].state = false;
      });

      const formikState = {
        errors,
        touched,
        values,
      };

      formikState.values.amenities[amenityIndex].configurationAmenityOptions[
        optionIndex
      ].configurationSubAmenities = configurationSubAmenities;

      setFormikState(formikState);
    };

  handleValidateOnSubmit = async () => {
    const { values, handleSubmit, setFieldValue, camperId } = this.props;

    let isValid = true;

    values.amenities.forEach((amenity) => {
      if (amenity.title === CAMPER_AMENITIES_CONFIG.ACCOMMODATION_TITLE) {
        const filtered = amenity.configurationSubAmenities.filter(
          (subAmenity) => isPresent(subAmenity.state) && subAmenity.state,
        );

        if (filtered.length === 0) {
          this.props.showMessage({
            messageSubTitle: {
              id: 'addNewCamper.amenities.accommodationRequired',
            },
            messageType: MESSAGE_TYPE.WARN,
          });

          isValid = false;
        }
      }

      amenity.configurationAmenityOptions?.forEach((option) => {
        if (option.state && option.configurationSubAmenities.length > 0) {
          const result = option.configurationSubAmenities.filter(
            (subAmenity) => subAmenity.state,
          );
          if (result.length === 0) {
            this.props.showMessage({
              messageSubTitle: {
                id: 'addNewCamper.amenities.amenityRequired',
              },
              messageType: MESSAGE_TYPE.WARN,
            });

            isValid = false;
          }
        }
      });
    });

    if (!values.redirectRoute) {
      const route = createRouteFromPathname(
        ROUTES.OWNER_DASHBOARD.EDIT_CAMPER.SPECIFICATIONS.PATH,
        null,
        { camper: camperId },
      );

      await setFieldValue('redirectRoute', route);
    }

    if (isValid) {
      handleSubmit();
    }

    return isValid;
  };

  onMaxAccommodationQuantityCallback = () => {
    this.props.showMessage({
      messageSubTitle: {
        id: 'addNewCamper.amenities.maxAccommodationNumber',
      },
      messageType: MESSAGE_TYPE.WARN,
    });
  };

  onMaxCustomAccommodationQuantityCallback = () => {
    this.props.showMessage({
      messageSubTitle: {
        id: 'validations.maxAllowedNumber',
      },
      messageType: MESSAGE_TYPE.WARN,
    });
  };

  get isLoading() {
    const { isLoadingAmenities, isLoadingHealthSafeties } = this.props;

    return isLoadingAmenities || isLoadingHealthSafeties;
  }

  render() {
    return (
      <AmenitiesComponent
        {...this.props}
        isLoading={this.isLoading}
        handleValidateOnSubmit={this.handleValidateOnSubmit}
        onMaxAccommodationQuantityCallback={
          this.onMaxAccommodationQuantityCallback
        }
        onMaxCustomAccommodationQuantityCallback={
          this.onMaxCustomAccommodationQuantityCallback
        }
        removeCustomAccommodation={this.removeCustomAccommodation}
        addCustomAccommodation={this.addCustomAccommodation}
        handleSwitchChange={this.handleSwitchChange}
        leavePagePrepare={this.leavePagePrepare}
      />
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  amenities: camperPossibleAmenitiesSelector(state),
  isLoadingAmenities: loadingSelector(
    state,
    createCamperAmenitiesEndpoint.endpoint,
  ),
  isLoadingHealthSafeties: loadingSelector(
    state,
    createCamperAmenityHealthSafetiesEndpoint.endpoint,
  ),
  isCamperExist: isCamperExistSelector(state, ownProps?.camperId),
  amenityHealthSafetyItems: amenityHealthAndSafetyItemsSelector(state),
});

const mapDispatchToProps = {
  onSubmit: createCamperAmenityHealthSafeties,
  showMessage,
  showModal: showModalAction,
  hideModal: hideModalAction,
};

export { Amenities as AmenitiesContainer };
export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withFormik({
    enableReinitialize: true,
    mapPropsToValues: ({ amenities, amenityHealthSafetyItems }) => ({
      amenities,
      amenityHealthSafetyItems,
    }),
    validationSchema: yup.object().shape({
      amenities: AMENITIES_VALIDATION_SCHEMA,
    }),
    handleSubmit: handleSubmitWithProps(['camperId']),
  }),
  withRouter,
)(Amenities);
