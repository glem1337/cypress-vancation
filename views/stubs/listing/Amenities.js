import {
  Row,
  Col,
  Switch,
  Checkbox,
  Divider,
  Button,
  Form,
} from 'antd';
import CarCard from './components/CarCard';
import BigCheckbox from './components/BigCheckbox';
import BigCheckboxCounter from './components/BigCheckboxCounter';
import TooltipIcon from '../shared/TooltipIcon';
import CustomAccBlock from './components/CustomAccBlock';
import LayoutListing from './LayoutListing';

const Amenities = () => (
  <LayoutListing withBackBtn>
    <>
      <Row>
        <Col span={24}>
          <h1 className="text-headline mb-8">
            Select your campers amenities
          </h1>
          <p className="mb-24">
            Not sure about all of the amenities on your campervan or RV?
            {/* eslint-disable-next-line react/no-unescaped-entities */}
            Don't worry, you can edit all of this information later!
          </p>
          <div className="d-flex align-items-center mb-24">
            <img
              src="/images/listing/amenities-svg/accommodation/couch.svg"
              className="mr-12"
              alt=""
            />
            <span className="text-subheader font-700">Accommodation</span>
          </div>
          <Row gutter={24}>
            <Col md={12}>
              <Form.Item>
                <BigCheckboxCounter
                  icon="accommodation/king_bed"
                  text="King Bed"
                  id="King-Bed"
                />
              </Form.Item>
            </Col>
            <Col md={12}>
              <Form.Item>
                <BigCheckboxCounter
                  icon="accommodation/queen_bed"
                  text="Queen Bed"
                  id="Queen-Bed"
                />
              </Form.Item>
            </Col>
            <Col md={12}>
              <Form.Item>
                <BigCheckboxCounter
                  icon="accommodation/twin_bed"
                  text="Twin Bed"
                  id="Twin-Bed"
                />
              </Form.Item>
            </Col>
            <Col md={12}>
              <Form.Item>
                <BigCheckboxCounter
                  icon="accommodation/single_bed"
                  text="Single Bed"
                  id="Single-Bed"
                />
              </Form.Item>
            </Col>
            <Col md={12}>
              <Form.Item>
                <BigCheckboxCounter
                  icon="accommodation/dinette_conversion"
                  text="Dinette Conversion"
                  id="Dinette-Conversion"
                />
              </Form.Item>
            </Col>
            <Col md={12}>
              <Form.Item>
                <BigCheckboxCounter
                  icon="accommodation/bunk_bed"
                  text="Bunk Bed"
                  id="Bunk-Bed"
                />
              </Form.Item>
            </Col>
          </Row>
          <Divider />
          <div className="mb-24 text-subheader font-400">
            Custom accommodation
          </div>
          <Row gutter={24}>
            <Col span={24}>
              <Row gutter={24}>
                <Col lg={12}>
                  <Form.Item>
                    <CustomAccBlock />
                  </Form.Item>
                </Col>
              </Row>
            </Col>
            <Col span={24}>
              <Row gutter={24}>
                <Col lg={12}>
                  <Form.Item>
                    <CustomAccBlock />
                  </Form.Item>
                </Col>
              </Row>
            </Col>
          </Row>
          <Button type="secondary">
            Add custom bed
          </Button>
          <Divider className="mt-24" />
          <div className="d-flex align-items-center mb-24">
            <img
              src="/images/listing/amenities-svg/kitchen/cooking_pot.svg"
              className="mr-12"
              alt=""
            />
            <span className="text-subheader font-700">Kitchen</span>
          </div>
          <Row gutter={24}>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="kitchen/stovetop" text="stovetop" id="stovetop" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="kitchen/oven" text="oven" id="oven" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="kitchen/sink" text="sink" id="sink" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="kitchen/hot-water" text="hot water" id="hot_water" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="kitchen/microwave" text="microwave" id="microwave" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="kitchen/fridge" text="fridge" id="fridge" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="kitchen/freezer" text="freezer" id="freezer" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="kitchen/cooler" text="cooler" id="cooler" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="kitchen/extra-countertop-space"
                  text="extra countertop space"
                  id="extra_countertop_space"
                />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <BigCheckbox icon="kitchen/dining-table" text="dining table" id="dining_table" />
            </Col>
          </Row>
          <Divider />
          <div className="d-flex align-items-center mb-24">
            <img
              src="/images/listing/amenities-svg/bathroom/bath.svg"
              className="mr-12"
              alt=""
            />
            <span className="text-subheader font-700">Bathroom</span>
          </div>
          <Row gutter={24}>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="bathroom/outdoor_shower" text="Outdoor Shower" id="outdoor_shower" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="bathroom/indoor_shower" text="indoor Shower" id="indoor_shower" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="kitchen/hot-water" text="hot water" id="hot_water" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="bathroom/toilet_urine_only" text="Toilet (Urine Only)" id="toilet_urine_only" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="bathroom/toilet_full_use" text="Toilet (Full Use)" id="toilet_full_use" />
              </Form.Item>
            </Col>
          </Row>
          <Divider />
          <div className="d-flex align-items-center mb-24">
            <img
              src="/images/listing/amenities-svg/climate_control/air_conditioning.svg"
              className="mr-12"
              alt=""
            />
            <span className="text-subheader font-700">Climate control</span>
          </div>
          <Row gutter={24}>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="climate_control/ceiling-fan"
                  text="ceiling fan"
                  id="ceiling_fan"
                />
              </Form.Item>
            </Col>
            <Col span={24}>
              <Form.Item>
                <div className="d-flex align-items-center mb-20">
                  <Switch />
                  <img
                    src="/images/listing/amenities-svg/climate_control/air-conditioner.svg"
                    className="mr-12 ml-12"
                    alt=""
                  />
                  <span>Air Conditioner</span>
                </div>
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="climate_control/hookups_required_for_a_c"
                  text="Hookups Required for A/C"
                  id="hookups_required_for_A/C"
                  tooltipPhrase="Hookups Required for A/C"
                />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="climate_control/off-grid-air-conditioner"
                  text="Off Grid Air Conditioner"
                  id="off_grid_air_conditioner"
                  tooltipPhrase="Hookups Required for A/C"
                />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="climate_control/a-c-only-while-camper-is-on"
                  text="A/C Only While Camper is On"
                  id="a-c-only-while-camper-is-on"
                />
              </Form.Item>
            </Col>
            <Col span={24}>
              <Form.Item>
                <div className="d-flex align-items-center mb-20">
                  <Switch />
                  <img
                    src="/images/listing/amenities-svg/climate_control/heating.svg"
                    className="mr-12 ml-12"
                    alt=""
                  />
                  <span>Heating</span>
                </div>
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="climate_control/portable-heater"
                  text="Portable Heater"
                  id="portable_heater"
                />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="climate_control/air-heating-system"
                  text="Air Heating System"
                  id="air_heating_system"
                  tooltipPhrase="Hookups Required for A/C"
                />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="climate_control/heating-only-while-camper-is-on"
                  text="Heating Only While Camper is On"
                  id="heating_only_while_camper_is_on"
                />
              </Form.Item>
            </Col>
          </Row>
          <Divider />
          <div className="d-flex align-items-center mb-24">
            <img
              src="/images/listing/amenities-svg/power_system/power_system.svg"
              className="mr-12"
              alt=""
            />
            <span className="text-subheader font-700">Power system</span>
          </div>
          <Row gutter={24}>
            <Col span={24}>
              <Form.Item>
                <div className="d-flex align-items-center mb-20">
                  <Switch />
                  <img
                    src="/images/listing/amenities-svg/power_system/solar-panels.svg"
                    className="mr-12 ml-12"
                    alt=""
                  />
                  <span>Off Grid Capable System</span>
                </div>
              </Form.Item>
            </Col>
            <Col span={24}>
              <Form.Item>
                <Checkbox className="mb-20">
                  <img
                    src="/images/listing/amenities-svg/power_system/electric-power-system.svg"
                    className="mr-12"
                    alt=""
                  />
                  Electric Power System
                </Checkbox>
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="power_system/battery-storage-under"
                  text="Battery Storage (under 300 amp hours)"
                  id="battery_storage_under"
                />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="power_system/battery-storage-plus"
                  text="Battery Storage (300+ amp hours)"
                  id="battery_storage_up"
                />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="power_system/alternator-charger"
                  text="Alternator Charger"
                  id="alternator_charger"
                  tooltipPhrase="Hookups Required for A/C"
                />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="power_system/inverter"
                  text="inverter"
                  id="inverter"
                  tooltipPhrase="Hookups Required for A/C"
                />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="power_system/electrical-outlets"
                  text="A/C Electrical Outlets"
                  id="electrical_outlets"
                  tooltipPhrase="Hookups Required for A/C"
                />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="power_system/solar-panels"
                  text="Solar Panels"
                  id="solar_panels"
                />
              </Form.Item>
            </Col>
            <Col span={24}>
              <Form.Item>
                <Checkbox className="mb-20">
                  <img
                    src="/images/listing/amenities-svg/power_system/propane-fuel-system.svg"
                    className="mr-12"
                    alt=""
                  />
                  Propane Fuel System
                </Checkbox>
              </Form.Item>
            </Col>
            <Col span={24}>
              <Form.Item>
                <Checkbox className="mb-20">
                  <img
                    src="/images/listing/amenities-svg/power_system/generator.svg"
                    className="mr-12"
                    alt=""
                  />
                  Generator
                </Checkbox>
              </Form.Item>
            </Col>
            <Col span={24}>
              <Form.Item>
                <div className="d-flex align-items-center mb-20">
                  <Switch />
                  <img
                    src="/images/listing/amenities-svg/power_system/shore-power-hookups.svg"
                    className="mr-12 ml-12"
                    alt=""
                  />
                  <span className="mr-8">Shore Power Hookups</span>
                  <TooltipIcon phrase="some som" iconClass="icon-info-f font-24" />
                </div>
              </Form.Item>
            </Col>
          </Row>
          <Divider />
          <div className="d-flex align-items-center mb-24">
            <img
              src="/images/listing/amenities-svg/bedroom/bed.svg"
              className="mr-12"
              alt=""
            />
            <span className="text-subheader font-700">Bedroom</span>
          </div>
          <Row gutter={24}>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="bedroom/large_mattress" text="Large Mattress" id="large_mattress" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="bedroom/extra_storage" text="Extra Storage" id="extra_storage" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox icon="bedroom/blackout_shades" text="Blackout Shades" id="blackout_shades" />
              </Form.Item>
            </Col>
            <Col md={8} lg={6}>
              <Form.Item>
                <BigCheckbox
                  icon="bedroom/sleep_sideways"
                  text="Sleep Sideways"
                  id="sleep_sideways"
                  tooltipPhrase="Hookups Required for A/C"
                />
              </Form.Item>
            </Col>
          </Row>
          <Divider />
          <div className="d-flex align-items-center mb-24">
            <img
              src="/images/listing/amenities-svg/entertainment/gamepad.svg"
              className="mr-12"
              alt=""
            />
            <span className="text-subheader font-700">Entertainment</span>
          </div>
          {/* TODO: When <CarCard /> is invisible, remove mb-16 */}
          <div className="mb-16">
            <Row gutter={24}>
              <Col md={8} lg={6}>
                <Form.Item>
                  <BigCheckbox
                    icon="entertainment/upgraded-speaker-system"
                    text="Upgraded Speaker System"
                    id="upgraded_speaker_system"
                  />
                </Form.Item>
              </Col>
              <Col md={8} lg={6}>
                <Form.Item>
                  <BigCheckbox
                    icon="entertainment/bluetooth-speaker"
                    text="Bluetooth Speaker"
                    id="bluetooth_speaker"
                  />
                </Form.Item>
              </Col>
              <Col md={8} lg={6}>
                <Form.Item>
                  <BigCheckbox
                    icon="entertainment/tv"
                    text="TV"
                    id="tv"
                  />
                </Form.Item>
              </Col>
              <Col md={8} lg={6}>
                <Form.Item>
                  <BigCheckbox
                    icon="entertainment/wifi-router"
                    text="Wifi Router"
                    id="wifi_router"
                  />
                </Form.Item>
              </Col>
              <Col md={8} lg={6}>
                <Form.Item>
                  <BigCheckbox
                    icon="entertainment/cell-signal-booster"
                    text="Cell Signal Booster"
                    id="cell_signal_booster"
                    tooltipPhrase="Hookups Required for A/C"
                  />
                </Form.Item>
              </Col>
              <Col md={8} lg={6}>
                <Form.Item>
                  <BigCheckbox
                    icon="entertainment/radio"
                    text="radio"
                    id="radio"
                  />
                </Form.Item>
              </Col>
            </Row>
          </div>
        </Col>
      </Row>
      <CarCard isFloat />
    </>
  </LayoutListing>
);

export default Amenities;
