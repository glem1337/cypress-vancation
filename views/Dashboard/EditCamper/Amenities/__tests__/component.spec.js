import { shallow } from 'enzyme';

import Amenities from '../component';

describe('Amenities component tests', () => {
  const props = {
    addCustomAccommodation: jest.fn(),
    removeCustomAccommodation: jest.fn(),
    handleSubmit: jest.fn(),
    handleSwitchChange: jest.fn(),
    handleValidateOnSubmit: jest.fn(),
    onMaxAccommodationQuantityCallback: jest.fn(),
    onMaxCustomAccommodationQuantityCallback: jest.fn(),
    leavePagePrepare: jest.fn(),
    isValid: true,
    isLoading: false,
    isCamperExist: true,
    values: {
      amenities: [
        {
          id: '1',
          title: 'Climate control',
          icon: '/images/listing/amenities-svg/climate_control/air_conditioning.svg',
          configurationSubAmenities: [
            {
              id: '1',
              title: 'Сeiling fan',
              icon: '/images/listing/amenities-svg/climate_control/ceiling-fan.svg',
              state: false,
            },
          ],
          configurationAmenityOptions: [
            {
              id: '1',
              title: 'Сeiling fan',
              icon: '/images/listing/amenities-svg/climate_control/ceiling-fan.svg',
              state: false,
              configurationSubAmenities: [
                {
                  id: '1',
                  title: 'Сeiling fan',
                  icon: '/images/listing/amenities-svg/climate_control/ceiling-fan.svg',
                  state: false,
                },
              ],
            },
          ],
        },
      ],
      amenityHealthSafetyItems: [
        {
          id: 'some id',
          title: 'some title',
          icon: 'some icon',
        },
      ],
    },
  };

  const component = shallow(<Amenities {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when camper amenities does not exist', () => {
    component.setProps({
      values: {
        amenities: undefined,
      },
    });

    expect(component).toMatchSnapshot();
  });
});
