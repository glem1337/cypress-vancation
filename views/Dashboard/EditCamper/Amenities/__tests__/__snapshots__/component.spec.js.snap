// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amenities component tests matches snapshot 1`] = `
<withRouter(EditCamper)
  canSave={true}
  hasFooter={true}
  isLoading={false}
  leavePageMethod={[MockFunction]}
  onSave={[MockFunction]}
>
  <div
    className="container"
  >
    <ForwardRef(InternalForm)>
      <Row
        gutter={24}
      >
        <Col
          span={24}
        >
          <h1
            className="text-headline mb-8"
          >
            <Memo(FormattedMessage)
              id="addNewCamper.amenities.title"
            />
          </h1>
          <AmenitySection
            handleSwitchChange={[MockFunction]}
            icon={null}
            index={0}
            items={
              Array [
                Object {
                  "icon": "/images/listing/amenities-svg/climate_control/ceiling-fan.svg",
                  "id": "1",
                  "state": false,
                  "title": "Сeiling fan",
                },
              ]
            }
            onMaxAccommodationQuantityCallback={[MockFunction]}
            options={
              Array [
                Object {
                  "configurationSubAmenities": Array [
                    Object {
                      "icon": "/images/listing/amenities-svg/climate_control/ceiling-fan.svg",
                      "id": "1",
                      "state": false,
                      "title": "Сeiling fan",
                    },
                  ],
                  "icon": "/images/listing/amenities-svg/climate_control/ceiling-fan.svg",
                  "id": "1",
                  "state": false,
                  "title": "Сeiling fan",
                },
              ]
            }
            title="Climate control"
            withDivider={false}
          />
          <Divider />
          <HealthAndSafetySection
            items={
              Array [
                Object {
                  "icon": "some icon",
                  "id": "some id",
                  "title": "some title",
                },
              ]
            }
          />
        </Col>
      </Row>
    </ForwardRef(InternalForm)>
  </div>
</withRouter(EditCamper)>
`;

exports[`Amenities component tests matches snapshot when camper amenities does not exist 1`] = `
<withRouter(EditCamper)
  canSave={true}
  hasFooter={true}
  isLoading={false}
  leavePageMethod={[MockFunction]}
  onSave={[MockFunction]}
>
  <div
    className="container"
  >
    <ForwardRef(InternalForm)>
      <Row
        gutter={24}
      >
        <Col
          span={24}
        >
          <h1
            className="text-headline mb-8"
          >
            <Memo(FormattedMessage)
              id="addNewCamper.amenities.title"
            />
          </h1>
          <Skeleton
            active={true}
            avatar={false}
            paragraph={true}
            title={true}
          />
        </Col>
      </Row>
    </ForwardRef(InternalForm)>
  </div>
</withRouter(EditCamper)>
`;
